### ISSUE TITLE
Failed generator installs should not finish with confirmative log

### ISSUE THREAD

**Issue by kontur** [yeoman/yo#226]
Hey,

when you try install a generator and yo fails to install the generator (in my case because it needs to run the npm installs with sudo) it still gives you a confirmative log, which is contradictory.

```
npm ERR! Please try running this command again as root/Administrator.
...
npm ERR! ...
npm ERR! ...
npm ERR! not ok code 0


I just installed a generator by running:

    npm install -g generator-angular
```

This is happening on yo 1.3.3 Mac OS X.

**Comment by addyosmani** [comment]
@SBoudrias Looks like we need to improve post-close behavior in https://github.com/yeoman/yo/blob/master/lib/routes/install.js#L130 to correct this. I agree we should only display the installation success message when we 100% know there hasn't been a failure somewhere else in the pipeline.

**Comment by SBoudrias** [comment]
Yeah. Also, maybe this should actually be fixed in yeoman-generator with a help/troubleshoot message if npmInstall fails.

**Comment by sindresorhus** [comment]
> Also, maybe this should actually be fixed in yeoman-generator with a help/troubleshoot message if npmInstall fails.

:+1:

**Comment by addyosmani** [comment]
:+1:
